iml-api/src/graphql/mod.rs

```diff
@@ -923,6 +966,138 @@ impl MutationRoot {
+                name,
+                ui_name,
+                ui_description,
+                managed,
+                worker,
+                user_selectable,
+                initial_state,
+                ntp,
+                corosync,
+                corosync2,
+                pacemaker,
+                "default"
+            )
+            VALUES
+            ($1, $2, $3, $4, $5, 'true', $6, $7, $8, $9, $10, 'false')
+            ON CONFLICT (name)
+            DO UPDATE
+            SET
+            ui_name = excluded.ui_name,
+            ui_description = excluded.ui_description,
+            managed = excluded.managed,
+            worker = excluded.worker,
+            user_selectable = excluded.user_selectable,
+            initial_state = excluded.initial_state,
+            ntp = excluded.ntp,
+            corosync = excluded.corosync,
+            corosync2 = excluded.corosync2,
+            pacemaker = excluded.pacemaker,
+            "default" = excluded.default
+        "#,
+            &profile.name,
+            &profile.ui_name,
+            &profile.ui_description,
+            &profile.managed,
+            &profile.worker,
+            &profile.initial_state,
+            &profile.ntp,
+            &profile.corosync,
+            &profile.corosync2,
+            &profile.pacemaker
+        )
+        .execute(&context.pg_pool)
+        .await?;
+
+        sqlx::query!(
+            r#"
+            INSERT INTO chroma_core_serverprofile_repolist (serverprofile_id, repo_id)
+            SELECT $1, repo_id
+            FROM UNNEST($2::text[])
+            AS t(repo_id)
+            ON CONFLICT DO NOTHING
+        "#,
+            &profile.name,
+            &repolist
+        )
+        .execute(&context.pg_pool)
+        .await?;
+
+        sqlx::query!(
+            r#"
+            INSERT INTO chroma_core_serverprofilepackage (package_name, server_profile_id)
+            SELECT package_name, $2
+            FROM UNNEST($1::text[])
+            as t(package_name)
+            ON CONFLICT DO NOTHING
+            "#,
+            &profile.packages.into_iter().collect::<Vec<_>>(),
+            &profile.name
+        )
+        .execute(&context.pg_pool)
+        .await?;
+
+        Ok(true)
+    }
+
+    #[graphql(arguments(profile_name(description = "Name of the profile to remove")))]
+    async fn remove_server_profile(
+        context: &Context,
+        profile_name: String,
+    ) -> juniper::FieldResult<bool> {
+        sqlx::query!(
+            "DELETE FROM chroma_core_serverprofile_repolist WHERE serverprofile_id = $1",
+            &profile_name
+        )
+        .execute(&context.pg_pool)
+        .await?;
+
+        sqlx::query!(
+            "DELETE FROM chroma_core_serverprofilepackage WHERE server_profile_id = $1",
+            &profile_name
+        )
+        .execute(&context.pg_pool)
+        .await?;
+
+        sqlx::query!(
+            "DELETE FROM chroma_core_serverprofile where name = $1",
```

Comment:
I think deletting from `chroma_core_serverprofile` will cause deletion from `chroma_core_serverprofile_repolist` and `chroma_core_serverprofilepackage` for the foreing keys. Not sure if we should rely on this.

Reply:
Well, repolist is internal Django table for many-to-many relation. package should get deleted as it's ON DELETE CASCADE. On the other hand, there are ServerProfileValidation and registrationtoken which aren't mentioned here, but depend on foreign key in serverprofile.
I'd say keeping referential integrity is DB's concern. @jgrund ?